pyrenew/convolve.py

```diff
@@ -219,43 +219,74 @@ def compute_delay_ascertained_incidence(
     return delay_obs_incidence
 
 
-def daily_to_epiweekly(
-    daily_value: ArrayLike,
-    first_dow: int = 0,
+def daily_to_weekly(
+    daily_values: ArrayLike,
+    input_data_first_dow: int = 0,
+    week_start_dow: int = 0,
 ):
     """
     Aggregate daily values (e.g.
     incident hospital admissions) into weekly total values.
 
     Parameters
     ----------
-    daily_value : ArrayLike
+    daily_values : ArrayLike
         Daily timeseries values (e.g. incident infections or incident ed visits).
-    first_dow : int
-        First day of the week in the input timeseries `daily_values`. An integer between 0 and 6, inclusive.
-        (0 for Monday, 6 for Sunday).
+    input_data_first_dow : int
+        First day of the week in the input timeseries `daily_values`.
+        An integer between 0 and 6, inclusive (0 for Monday, 6 for Sunday).
         If first_dow is not 0, the incomplete first
-        epiweek is ignored and epiweekly values
-        starting from second week is returned.
-        Defaults to 0.
+        week is ignored and weekly values starting
+        from second week is returned. Defaults to 0.
+    week_start_dow : int
+        The desired starting day of the week for the output weekly aggregation.
+        An integer between 0 and 6, inclusive. Defaults to 0 (Monday).
 
     Returns
     -------
     ArrayLike
-        Data converted to epiweekly values starting
-        with the first full epiweek available.
+        Data converted to weekly values starting
+        with the first full week available.
     """
-    if first_dow < 0 or first_dow > 6:
-        raise ValueError("First day of the week must be between 0 and 6")
+    if input_data_first_dow < 0 or input_data_first_dow > 6:
+        raise ValueError("First day of the week must be between 0 and 6.")
 
-    if first_dow > 0:
-        daily_value = daily_value[7 - first_dow :]
+    if week_start_dow < 0 or week_start_dow > 6:
+        raise ValueError(
+            "First day of aggregated data must be between 0 and 6."
+        )
+
+    if input_data_first_dow != week_start_dow:
+        offset = (week_start_dow - input_data_first_dow) % 7
+        daily_values = daily_values[offset:]
```

Comment:
I don't think you need the if, since an offset of 0 should work as expected, no?

Reply:
Right, will remove the condition!